handler/src/main/java/io/netty/handler/ssl/ReleasableOpenSslContext.java

```diff
@@ -0,0 +1,782 @@
+     * At this point {@code 0} will be returned.
+     *
+     * @deprecated use {@link #sslCtxPointer()}
+     */
+    @Deprecated
+    public final long context() {
+        return ctx;
+    }
+
+    /**
+     * Returns the stats of this context.
+     *
+     * @deprecated use {@link #sessionContext#stats()}
+     */
+    @Deprecated
+    public final OpenSslSessionStats stats() {
+        return sessionContext().stats();
+    }
+
+    /**
+     * Specify if remote initiated renegotiation is supported or not. If not supported and the remote side tries
+     * to initiate a renegotiation a {@link SSLHandshakeException} will be thrown during decoding.
+     */
+    public void setRejectRemoteInitiatedRenegotiation(boolean rejectRemoteInitiatedRenegotiation) {
+        this.rejectRemoteInitiatedRenegotiation = rejectRemoteInitiatedRenegotiation;
+    }
+
+    /**
+     * Sets the SSL session ticket keys of this context.
+     *
+     * @deprecated use {@link OpenSslSessionContext#setTicketKeys(byte[])}
+     */
+    @Deprecated
+    public final void setTicketKeys(byte[] keys) {
+        sessionContext().setTicketKeys(keys);
+    }
+
+    @Override
+    public abstract OpenSslSessionContext sessionContext();
+
+    /**
+     * Returns the pointer to the {@code SSL_CTX} object for this {@link ReleasableOpenSslContext}.
+     * Be aware that it is freed as soon as the {@link #release()}  method is called.
+     * At this point {@code 0} will be returned.
+     */
+    public final long sslCtxPointer() {
+        return ctx;
+    }
+
+    // IMPORTANT: This method must only be called from either the constructor or the finalizer as a user MUST never
+    //            get access to an OpenSslSessionContext after this method was called to prevent the user from
+    //            producing a segfault.
+    final void destroy() {
+        synchronized (ReleasableOpenSslContext.class) {
+            if (ctx != 0) {
+                SSLContext.free(ctx);
+                ctx = 0;
+            }
+
+            // Guard against multiple destroyPools() calls triggered by construction exception and finalize() later
+            if (aprPool != 0) {
+                Pool.destroy(aprPool);
+                aprPool = 0;
+            }
+        }
+    }
+
+    protected static X509Certificate[] certificates(byte[][] chain) {
+        X509Certificate[] peerCerts = new X509Certificate[chain.length];
+        for (int i = 0; i < peerCerts.length; i++) {
+            peerCerts[i] = new OpenSslX509Certificate(chain[i]);
+        }
+        return peerCerts;
+    }
+
+    protected static X509TrustManager chooseTrustManager(TrustManager[] managers) {
+        for (TrustManager m : managers) {
+            if (m instanceof X509TrustManager) {
+                return (X509TrustManager) m;
+            }
+        }
+        throw new IllegalStateException("no X509TrustManager found");
+    }
+
+    protected static X509KeyManager chooseX509KeyManager(KeyManager[] kms) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)